### ISSUE TITLE
Excluding folder does not affect (or incorrectly affects) browse path (was #1021)

### ISSUE THREAD

**Issue by Gruntfuggly** [microsoft/vscode-cpptools#1066]
As per title...

with "C_Cpp.intelliSenseEngine" set to "Tag Parser" and

from c_cpp_properties.json
```            
"browse": {
                "path": [
                    "/usr/include/c++/5",
                    "/usr/include/x86_64-linux-gnu/c++/5",
                    "/usr/local/include",
                    "/usr/include",
                    "/usr/include/x86_64-linux-gnu",
                    "${workspaceRoot}",
                    "/home/nscott/workspace/tmb/"
                ],
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": ""
            }
```

from User setting:
```
    "files.exclude": {
        "**/.DS_Store": true,
        "**/.git": true,
        "**/.hg": true,
        "**/.svn": true,
        "**/gen": true,
        "**/Entries": true,
        "**/Makefile*": true,
        "**/moc_*": true,
        "**/bin": true,
        "**/.obj": true,
        "**/.moc": true,
        "**/libs": true,
        "**/tmm/asl0632": true,
        "**/daemons/tmm/include/*": true
    },
```

Note the last entry. Go to definition takes me to a file in 
`
~/workspace/tmb/daemons/tmm/include/pdmconv
`

The correct file is actually in 
`
~/workspace/tmb/tmm/bricks/pdmconv
`

Sometimes it shows me both and allows me to choose, but after resetting the intellisense database it only takes me to the incorrect file.

**Comment by sean-mcmanus** [comment]
I'm not able to repro this bug on Linux. Do you see the excluded path being removed from the VS Code Explorer view? Does the symbol search find the excluded file or global variables in the excluded file?

Can you enable logging? We have logging about the paths/files that are excluded or not: https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/LanguageServer/Enabling%20logging.md . Also, set your log level to 7 (actually, some exclude logging is at level 9 too).

**Comment by Gruntfuggly** [comment]
The folder is not removed from the Explorer view because it's outside of the workspace - could that be part of the problem? Also, Go to symbol in workspace still finds the file.

**Comment by bobbrow** [comment]
Which folder maps to `${workspaceRoot}`?  I had assumed it was `~/workspace`.  It's possible that the pattern matching isn't working for files outside of the workspace root.

**Comment by sean-mcmanus** [comment]
Yeah, files.exclude is only applying to the current workspace. I remember we had a bug on this a long time ago, but we never fixed it.

**Comment by sean-mcmanus** [comment]
Fixed with 0.18.0.
